main.go

```diff
@@ -360,8 +364,18 @@ func NewDiffTool(legacyMode bool) (*xdcrDiffTool, error) {
 		logCtx.SetLogLevel(xdcrLog.LogLevelDebug)
 		gocb.SetLogger(gocb.VerboseStdioLogger())
 	}
-
-	difftool.selfRef, _ = metadata.NewRemoteClusterReference("", base.SelfReferenceName, options.sourceUrl, options.sourceUsername, options.sourcePassword,
+	var poolsInfo map[string]interface{}
+	var sourceClusterUUID string
+	err, statusCode := difftool.utils.QueryRestApi(staticHostAddr(), xdcrBase.PoolsPath, false, xdcrBase.MethodGet, "", nil, 0, &poolsInfo, difftool.logger)
+	if err != nil || statusCode != 200 {
+		return nil, fmt.Errorf("Failed on calling %v, err=%v, statusCode=%v\n", xdcrBase.PoolsPath, err, statusCode)
+	}
+	uuidObj, ok := poolsInfo[xdcrBase.RemoteClusterUuid]
```

Comment:
Add a comment indicating that `RemoteClusterUuid` is just purely "uuid" and can be used for local cluster UUID